### PR TITLE
[OAuth2] Fix storing empty refresh token on refreshToken call.

### PR DESCRIPTION
**Pull request by Hilbrand**
When a call is made for a new access token it might not return the refresh token in the data.
This was the case with Spotify Api, only to find it out after 2 hours running. Because the first save is from the authorization and that contains the refresh token. The first call to refresh the access token (after 1 hour, expire time is 3600 sec.) sets the refresh token to null. And then after access token expires again (2nd hour) it gives the exception no refresh token available.

Some minor changes: Moved creating OAuthConnector after checks which throw exceptions.

Signed-off-by: Hilbrand Bouwkamp <hilbrand@h72.nl>

### CODE REVIEW

File: bundles/auth/org.eclipse.smarthome.auth.oauth2client/src/main/java/org/eclipse/smarthome/auth/oauth2client/internal/OAuthClientServiceImpl.java [hunk 1]
```diff
@@ -26,6 +26,7 @@
 import org.eclipse.jdt.annotation.NonNull;
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.util.StringUtil;
```
Comment (kaikreuzer) [inline bundles/auth/org.eclipse.smarthome.auth.oauth2client/src/main/java/org/eclipse/smarthome/auth/oauth2client/internal/OAuthClientServiceImpl.java]: Are you sure you want to use such a util class from Jetty? Maybe rather go for Apache Commons Lang...

File: bundles/auth/org.eclipse.smarthome.auth.oauth2client/src/main/java/org/eclipse/smarthome/auth/oauth2client/internal/OAuthClientServiceImpl.java [hunk 2]
```diff
@@ -26,6 +26,7 @@
...
 import org.eclipse.jetty.util.UrlEncoded;
```
Comment (kaikreuzer) [inline bundles/auth/org.eclipse.smarthome.auth.oauth2client/src/main/java/org/eclipse/smarthome/auth/oauth2client/internal/OAuthClientServiceImpl.java]: I know, this is not part of this PR, but I was just wondering, why we have a Jetty dependency here - could `URLEncoder` be used instead?

File: bundles/auth/org.eclipse.smarthome.auth.oauth2client/src/main/java/org/eclipse/smarthome/auth/oauth2client/internal/OAuthClientServiceImpl.java (hunk 1 above)
Comment (Hilbrand) [reply bundles/auth/org.eclipse.smarthome.auth.oauth2client/src/main/java/org/eclipse/smarthome/auth/oauth2client/internal/OAuthClientServiceImpl.java]: I used it because that package is already used and the apache is not, so it would add a new library.

File: bundles/auth/org.eclipse.smarthome.auth.oauth2client/src/main/java/org/eclipse/smarthome/auth/oauth2client/internal/OAuthClientServiceImpl.java (hunk 2 above)
Comment (Hilbrand) [reply bundles/auth/org.eclipse.smarthome.auth.oauth2client/src/main/java/org/eclipse/smarthome/auth/oauth2client/internal/OAuthClientServiceImpl.java]: `URLEncoder` returns a complete string. Here is seems `UrlEncoded` is used to get easy access to parts of the url.
